src/Symfony/Component/HtmlSanitizer/HtmlSanitizer.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer;
+
+use Symfony\Component\HtmlSanitizer\Parser\MastermindsParser;
+use Symfony\Component\HtmlSanitizer\Parser\ParserInterface;
+use Symfony\Component\HtmlSanitizer\Reference\W3CReference;
+use Symfony\Component\HtmlSanitizer\TextSanitizer\StringSanitizer;
+use Symfony\Component\HtmlSanitizer\Visitor\DomVisitor;
+
+/**
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @experimental
+ */
+final class HtmlSanitizer implements HtmlSanitizerInterface
+{
+    private HtmlSanitizerConfig $config;
+    private int $maxInputLength;
+    private ParserInterface $parser;
+
+    /**
+     * @var DomVisitor[]
+     */
+    private array $domVisitors = [];
+
+    public function __construct(HtmlSanitizerConfig $config, int $maxInputLength = 20000, ParserInterface $parser = null)
+    {
+        $this->config = clone $config;
+        $this->maxInputLength = $maxInputLength;
+        $this->parser = $parser ?? new MastermindsParser();
+    }
+
+    public function sanitize(string $input): string
+    {
+        return $this->sanitizeWithContext(W3CReference::CONTEXT_BODY, $input);
+    }
+
+    public function sanitizeFor(string $element, string $input): string
+    {
+        return $this->sanitizeWithContext(W3CReference::CONTEXTS_MAP[$element] ?? W3CReference::CONTEXT_BODY, $input);
```

Comment:
not sure of the fallback to BODY since there is already a method for that. shouldn't we throw instead?

Reply:
Well, most elements don't have special rules for that and need the same rule than body. this logic here avoids making `CONTEXTS_MAP` huge.

Reply:
Makes sense thanks